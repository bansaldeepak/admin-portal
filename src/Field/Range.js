import React from 'react';

import { makeStyles } from '@material-ui/core/styles';

import Slider from '@material-ui/core/Slider';

const useStyles = makeStyles((theme) => ({
  textField: {
    minWidth: '100px'
  }
}));

const EnhancedRange = (props) => {
  const classes = useStyles();
  const { key, name, min, step, max, value, formData, handleChange } = props;

  return (
    <div className={classes.textField}>
      <Slider
        key={key}
        name={name}
        defaultValue={value || 0}
        min={(typeof min === 'function') ? min(formData) : min}
        max={(typeof max === 'function') ? max(formData) : max}
        step={(typeof step === 'function') ? step(formData) : step}
        onChange={(event, value) => {
          handleChange(name, value);
        }}
        valueLabelDisplay='auto'
        marks={[{value: min, label: min}, {value: max, label: max}]}
      />
    </div>
  );
}

export default EnhancedRange;
