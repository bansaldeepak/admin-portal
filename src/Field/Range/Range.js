import React from "react";

import { makeStyles } from "@material-ui/core/styles";

import Slider from "@material-ui/core/Slider";

const useStyles = makeStyles((theme) => ({
  textField: {
    minWidth: "100px",
  },
}));

const EnhancedRange = (props) => {
  const classes = useStyles();
  const { key, name, min, step, max, value, fieldValues, handleChange } = props;

  // const { decimal } = props;
  // if (type === 'range') {
  //   value = (isNaN(value) || typeof value === 'string') ? value : value.toFixed(decimal || 2);
  // }

  return (
    <div className={classes.textField}>
      <Slider
        key={key}
        name={name}
        defaultValue={value || 0}
        min={typeof min === "function" ? min(fieldValues) : min}
        max={typeof max === "function" ? max(fieldValues) : max}
        step={typeof step === "function" ? step(fieldValues) : step}
        onChange={(event, value) => {
          handleChange(value);
        }}
        valueLabelDisplay="auto"
        marks={[
          { value: min, label: min },
          { value: max, label: max },
        ]}
      />
    </div>
  );
};

export default EnhancedRange;
