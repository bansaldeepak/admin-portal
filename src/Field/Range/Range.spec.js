import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallowToJson } from "enzyme-to-json";
import Range from "./Range";

configure({ adapter: new Adapter() });

describe("Range", () => {
  it("Render Range with default options", () => {
    const component = shallow(<Range />);
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
