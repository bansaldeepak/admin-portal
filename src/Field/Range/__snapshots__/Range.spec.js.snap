// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range Render Range with default options 1`] = `
<div
  className="makeStyles-textField-1"
>
  <WithStyles(ForwardRef(Slider))
    defaultValue={0}
    marks={
      Array [
        Object {
          "label": undefined,
          "value": undefined,
        },
        Object {
          "label": undefined,
          "value": undefined,
        },
      ]
    }
    onChange={[Function]}
    valueLabelDisplay="auto"
  />
</div>
`;
